### PR TITLE
Patch/gamemode entities

### PR DESCRIPTION
**Pull request by DoctorGurke**
I've just quickly patched up how gamemode entities link together. The orb now has a proper particle system that'll get spawned in-game and show up in-hammer and it references a gamemode entity via its targetname.

Do note that for some reason the gamemode entity targetname gets automatically capitalized in-game, so there's a temp hack with ToLower() on the linking process. It should be fine though, as long as we don't have SUPER ambiguous gamemode entity names that only differ in capitalization.

### CODE REVIEW

File: code/Base/GamemodeSystem/GamemodeOrb.cs [hunk 1]
```diff
@@ -4,31 +4,27 @@ namespace Sports;
...
+
 	protected TimeSince LastTouch = 1f;
 
 	[Event.Entity.PostSpawn]
 	protected void PostEntitiesSpawned()
 	{
-		LinkedGamemode = SportsGame.Instance?.GetGamemodeFromId( GamemodeId );
-		Log.Debug( "Orb: Linked to gamemode." );
+		LinkedGamemode = SportsGame.Instance?.GetGamemodeFromId( Gamemode );
+		if ( LinkedGamemode is not null )
```
Comment (DevulTj) [inline code/Base/GamemodeSystem/GamemodeOrb.cs:26]: I would prefer using IsValid for Entities. It's just something I've done everywhere else. What's your take on this?

File: code/Base/GamemodeSystem/GamemodeOrb.cs [hunk 2]
```diff
@@ -64,8 +59,19 @@ public override void StartTouch( Entity other )
 
 		if ( other is PlazaPlayer player )
 		{
+			Log.Debug( $"Orb trying to add player: {player.Client.Name} to gamemode: {Gamemode}" );
 			LinkedGamemode?.AddClient( player.Client );
 			LastTouch = 0;
 		}
 	}
+
+	[Event.Tick]
```
Comment (DevulTj) [inline code/Base/GamemodeSystem/GamemodeOrb.cs:68]: Take note that this will tick on the server & client at the same time.

**Review by DevulTj — CHANGES_REQUESTED** [review]

File: code/Base/GamemodeSystem/GamemodeOrb.cs (hunk 2 above)
Comment (DoctorGurke) [reply code/Base/GamemodeSystem/GamemodeOrb.cs:68]: this was an oversight as I just wanted to cram debug draw in while the stupid particle wasn't showing haha, will adjust it

File: code/Base/GamemodeSystem/GamemodeOrb.cs (hunk 1 above)
Comment (DoctorGurke) [reply code/Base/GamemodeSystem/GamemodeOrb.cs:26]: I'm fine with that, IsValid does a null check anyways, so it's probably safer to use in the sbox context

**Pull request merged**